### PR TITLE
For #8156: Add indicator to synced tabs suggestion in awesomebar

### PR DESCRIPTION
**Pull request by hakkikaancaliskan**
Fenix side: https://github.com/mozilla-mobile/fenix/pull/13923

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/SyncedTabsStorageSuggestionProvider.kt [hunk 1]
```diff
@@ -16,10 +17,14 @@ import java.util.UUID
...
  */
+@Suppress("LongParameterList")
 class SyncedTabsStorageSuggestionProvider(
     private val syncedTabs: SyncedTabsStorage,
     private val loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
-    private val icons: BrowserIcons? = null
+    private val icons: BrowserIcons? = null,
+    private val indicatorDesktop: Drawable? = null,
+    private val indicatorMobile: Drawable? = null,
+    private val indicatorTablet: Drawable? = null
```
Comment (jonalmeida) [inline components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/SyncedTabsStorageSuggestionProvider.kt:27]:
```suggestion
    private val icons: BrowserIcons? = null,
    private val deviceIndicators: DeviceIndicators = DeviceIndicators()
```

How about we turn this into a data class of indicators instead?

```kotlin
data class DeviceIndicators(
    val desktop: Drawable? = null,
    val mobile: Drawable? = null,
    val tablet: Drawable? = null,
)
```

**Review by jonalmeida — COMMENTED** [review]
Thanks for the change! Let's also add some tests and a changelog entry. 🙂

File: components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/SyncedTabsStorageSuggestionProvider.kt (hunk 1 above)
Comment (hakkikaancaliskan) [reply components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/SyncedTabsStorageSuggestionProvider.kt:27]: Ohh i didn't know we can do like that. Great!
I'm not good with tests but i'll try my best.

Comment (jonalmeida) [reply components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/SyncedTabsStorageSuggestionProvider.kt:27]: Have a look at other tests in the codebase, and how `SyncedTabsStorageSuggestionProviderTest` can also help give some ideas on how to create more tests there.

Feel free to ping me if you need more help! :)

**Comment by hakkikaancaliskan** [comment]
@jonalmeida because of indicatorIcons come from fenix side, i'm unable to test them because of they're null.
Also when i try to mock them via `whenever` it failse because there is no method returning indicatorIcons.
I've added tests via flags but not with indicatorIcons.
How can i achieve this?

**Comment by hakkikaancaliskan** [comment]
Also when you finish review, let me know and i'll rebase commits :)

File: components/concept/awesomebar/src/main/java/mozilla/components/concept/awesomebar/AwesomeBar.kt [hunk 2]
```diff
@@ -122,7 +122,10 @@ interface AwesomeBar {
         enum class Flag {
             BOOKMARK,
             OPEN_TAB,
-            CLIPBOARD
+            CLIPBOARD,
+            SyncedTabsDesktop,
+            SyncedTabsMobile,
+            SyncedTabsTablet
```
Comment (jonalmeida) [inline components/concept/awesomebar/src/main/java/mozilla/components/concept/awesomebar/AwesomeBar.kt:128]:
```suggestion
            CLIPBOARD,
            SYNC_TAB
```

I don't think we need to separate each type here for the flag. Let's just use one and match the styling.

File: docs/changelog.md [hunk 3]
```diff
@@ -28,7 +28,8 @@ permalink: /changelog/
       * BUGFIX: Support `HashSet` and `Dictionary` in the C# generated code.
     * Add a 10MB quota to the pending pings storage. ([#1100](https://github.com/mozilla/glean/pull/1110))
     * Handle ping registration off the main thread. This removes a potential blocking call ([#1132](https://github.com/mozilla/glean/pull/1132)).
-
+* **feature-syncedtabs**
+  * Added support for adding indicator to synced tabs awesomebar suggestions.
```
Comment (jonalmeida) [inline docs/changelog.md:32]: nit
```suggestion
  * Added support for indicators to synced tabs `AwesomeBar` suggestions.
```

**Review by jonalmeida — COMMENTED** [review]
Looks good!

After these changes I think you can squash your changes into one. We also follow the commit format "Closes #[issue_number]: Description of the issue in a short sentence".

**Review by jonalmeida — APPROVED** [review]
bors r+

Looks great, thanks!

**Pull request merged**